framework/src/packets/ethernet.rs

```diff
@@ -28,18 +30,25 @@ use std::fmt;
 /// MAC address
 #[derive(Default, Debug, Copy, Clone, PartialEq, Eq)]
 #[repr(C, packed)]
-pub struct MacAddr(pub [u8; 6]);
+pub struct MacAddr([u8; 6]);
 
 impl MacAddr {
     pub const UNSPECIFIED: Self = MacAddr([0, 0, 0, 0, 0, 0]);
 
+    #[allow(clippy::many_single_char_names)]
     pub fn new(a: u8, b: u8, c: u8, d: u8, e: u8, f: u8) -> Self {
         MacAddr([a, b, c, d, e, f])
     }
 
     pub fn new_from_slice(slice: &[u8]) -> Self {
         MacAddr([slice[0], slice[1], slice[2], slice[3], slice[4], slice[5]])
     }
+
+    /// Returns the six bytes the MAC address consists of
+    #[allow(clippy::trivially_copy_pass_by_ref)]
```

Comment:
had to turn this off. clippy is being annoying here. seems like a lint bug.

Reply:
np. find to do that from time to time.